src/qtt/measurements/scans.py

```diff
@@ -870,22 +870,41 @@ def _convert_scanjob_vec(self, station, steplength=None, sweeplength=None, stepv
         """
 
         def _update_step_size_from_length(data, length):
-            """ From desired number of data points determine the required step size """
+            """ From desired number of data points determine the required step size.
+            length > 1: to get 'inclusive' data with length = 1, start and end values must be the same and step value
+            must be 0 which are invalid values.
+            """
             if length is not None:
                 if length == 1:
-                    data['step'] = 0
+                    raise ValueError('length must be > 1')
                 else:
                     if 'range' in sweepdata:
                         data['step'] = data['range'] / (length-1)
                     else:
                         data['step'] = (data['end'] - data['start']) / (length-1)
 
-        def _calculate_sweepvalues(param, sweep_data) -> SweepFixedValues:
-            """ From parameter and sweep data create a SweepFixedValues object """
-            sweep_values = SweepFixedValues(param,
-                                            start=sweep_data['start'],
-                                            stop=sweep_data['end'],
-                                            step=sweep_data['step'])
+        def _calculate_sweepvalues(param, sweep_data, inclusive_end=False) -> SweepFixedValues:
+            """ From parameter and sweep data create a SweepFixedValues object. """
+            if sweep_data['step'] == 0.:
+                raise ValueError('step value may not be 0')
+            steps = abs((sweep_data['end'] - sweep_data['start']) / sweep_data['step'])
+            tolerance = 1e-10
+            steps_lo = int(np.floor(steps + tolerance))
+            steps_hi = int(np.ceil(steps - tolerance))
+
+            if inclusive_end:
+                if steps_lo != steps_hi:
+                    sweep_data['end'] = sweep_data['start'] + steps_lo * sweep_data['step']
+
+                sweep_values = SweepFixedValues(param,
+                                                start=sweep_data['start'],
+                                                stop=sweep_data['end'],
```

Comment:
```suggestion
                                                stop=sweep_data['end'] + .5*sweep_data['step'],
```
To avoid any rounding errors? Not sure whether they can occur

Reply:
Adding 0.5*step might not be a good idea, since qcodes determines the number of steps with the same code as lines 890-893 and raises an exception when steps_lo != steps_hi, a condition that is likely to occur when 0.5*step is added.

Reply:
Yes, you are right Sander. With adjusting the 'end' parameter in the previous lines I assured that there was never a rounding error in the SweepFixValue creation.